Code.Rmd

```diff
@@ -29,23 +42,126 @@ Congratulations!
 You've learned your first package development workflow.
 Even if you learn nothing else from this book, you'll have gained a useful workflow for editing and reloading R code.
 
-## Organising your functions
+## Organising your functions into files {#r-organising}
+
+*We originally decided to remove this in deference to material in <https://style.tidyverse.org>. The removed text is in [tidyverse/style/#121](https://github.com/tidyverse/style/issues/121). In hindsight, I think removal is too extreme and I've re-introduced this section. First, the original material really isn't in style.tidyverse.org (yet?). Second, people really want guidance on this and it feels misguided to omit from this book out of some adherence to the DRY principle.*
+
+The only hard rule is that your package should store its function definitions in R scripts, i.e. files with extension `.R`, that live in the `R/` directory[^no-subdirs].
+However, a few more conventions can make the source code of your package easier to navigate and relieve you of re-answering "How should I name this?" each time you create a new file[^file-style].
+
+[^no-subdirs]: Unfortunately you can't use subdirectories inside `R/`.
+The next best thing is to use a common prefix, e.g., `abc-*.R`.
+
+[^file-style]: More advice about naming and organising R files is in The Tidyverse Style Guide.
+General advice is in <https://style.tidyverse.org/files.html> and some package-specific advice is in <https://style.tidyverse.org/package-files.html>.
+
+The file name should be meaningful and convey which functions are defined within.
+While you're free to arrange functions into files as you wish, the two extremes are bad: don't put all functions into one file and don't put each function into its own separate file.
+This advice should inform your general policy, but there are exceptions to every rule.
+If a specific function is very large or has lots of documentation, it can make sense to give it its own file, named after the function.
+More often, a single `.R` file will contain multiple function definitions: such as a main function and its supporting helpers, a family of related functions, or some combination of the two.
+
+Here are some examples from the actual source of the [tidyr package](http://tidyr.tidyverse.org/):
+
+```{r include = FALSE}
+library(tidyverse)
+
+# TODO: should these links be SHA'd to guard against change?
```

Comment:
Yes, especially since they make me see that I don't like the current organisation